### PR TITLE
Add new build scripts

### PR DESCRIPTION
**Pull request by alschmiedt**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Proposed Changes
Add: 
npm run build:compressed
npm run build:core
npm run build:generators

### Reason for Changes
Makes it more in sync with what we used to have.

### Test Coverage

Tested on:

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -440,7 +468,7 @@ gulp.task('build-langfiles', function(done) {
  *     msg/json/*.js
  */
 gulp.task('build', gulp.parallel(
-  'build-core',
+  'build-compressed',
```
Comment (samelhusseini) [inline gulpfile.js]: Maybe call build-core and build-generators from here?

File: package.json [hunk 2]
```diff
@@ -20,7 +20,9 @@
     "build": "gulp build",
     "build:blocks": "gulp build-blocks",
     "build:core": "gulp build-core",
-    "build:debug": "gulp build-core --verbose --strict",
+    "build:compressed":"gulp build-compressed",
+    "build:generators": "gulp build-generators",
```
Comment (samelhusseini) [inline package.json]: Sort.

File: package.json [hunk 3]
```diff
@@ -20,7 +20,9 @@
...
+    "build:debug": "gulp build:compressed --verbose --strict",
```
Comment (samelhusseini) [inline package.json]: this should be ``gulp build-compressed --verbose --strict``

**Comment by samelhusseini** [comment]
There are few more references to build-core around the place, namely: 
- ``gulp build-core --closure-library`` in requires.js that should be replaced with build-compressed
- ``gulp build-core`` in check_metadata.sh that should be replaced with build-compressed

File: package.json (hunk 2 above)
Comment (alschmiedt) [reply package.json]: done.

File: gulpfile.js (hunk 1 above)
Comment (alschmiedt) [reply gulpfile.js]: done

File: package.json (hunk 2 above)
Comment (alschmiedt) [reply package.json]: done

File: package.json (hunk 3 above)
Comment (alschmiedt) [reply package.json]: done

**Pull request merged**